332 - Navigation menu

klinki opened the pull request:
Rebased branch with almost no refactorings. 
I removed parts where I changed the way how data are loaded for existing components.

klinki commented:
Sorry for creating another PR, I didn't intend to, I wanted to close the original PR https://github.com/windup/windup-web/pull/86 until I finish rebasing (to prevent unnecessary builds) and reopen it again. But Github doesn't allow to reopen closed PR after rebase https://github.com/isaacs/github/issues/361.

klinki commented:
retest

jsight commented:
retest

jsight commented:
A couple of notes:
 - Settings should link to the analysis context settings instead of to the global configuration
 - Run windup doesn't provide any user feedback that windup is running. This doesn't necessarily have to be addressed in this PR, though, as I think it is a more general problem.
 - Once you have navigated to a particular group, there is no way to get back to the group list

I think the first one is the main issue for this to get merged. Also, I have noticed that the header on the project list and group list pages seems to be partially hidden by the navbar.

Overall, this looks good, though. I think that we should discuss adding a few more stories to the spring tomorrow to get the flow from groups to individual reports (and a breadcrumbs bar) farther along.

jsight commented:
A couple more notes:
 - The link to "Run Windup" doesn't do anything.
 - I think the links to things that don't exist (Dependencies and Dashboard) should be removed for now
 - MINOR: The spacing on the list pages still seems a little off. For example, the "Project List" text at the top looks really close to the navbar: http://i.imgur.com/zvMwhT6.png

The bar itself looks really nice, though. I think this will be ready to merge with these minor changes.

klinki commented:
Well the Run Windup link actually executed windup, but it didn't show any notification. I added notification for now. I think there is also a story to add execution progress bar to group page as well.

I fixed that spacing issue. I noticed we didn't use some css classes which are used in patternfly examples. So I added them. Spacing might look little different now. But it should look better.

klinki commented:
retest

jsight commented:
@klinki - Interesting... it didn't seem to be kicking off analysis. I wasn't too worried about showing progress, as I remember that new story. I will try it again today.

jsight commented:
Looks good to me... thanks!
